src/register/RegistrationPage.jsx

```diff
@@ -317,11 +317,22 @@ class RegistrationPage extends React.Component {
   }
 
   handleOnChange = (e) => {
-    if (!(e.target.name === 'username' && e.target.value.length > 30)) {
-      this.setState({
-        [e.target.name]: e.target.value,
-      });
+    const { name, value } = e.target;
+    if (name === 'username') {
+      if (value.length > 30) {
+        return;
+      }
+      if (value.startsWith(' ')) {
+        this.setState({
+          [name]: value.substring(1),
+        });
+        return;
+      }
     }
+
+    this.setState({
+      [name]: value,
+    });
```

Comment:
Looks good to me, just a suggestion if we can have setState in one place only? like this
```suggestion
      if (value.startsWith(' ')) {
          value = value.substring(1),
      }
    }

    this.setState({
      [name]: value,
    });
```